jsonnet/kube-prometheus/components/kube-state-metrics.libsonnet

```diff
@@ -111,6 +111,30 @@ function(params) (import 'github.com/kubernetes/kube-state-metrics/jsonnet/kube-
     image: ksm._config.kubeRbacProxyImage,
   }),
 
+  networkPolicy: {
+    apiVersion: 'networking.k8s.io/v1',
+    kind: 'NetworkPolicy',
+    metadata: ksm.service.metadata,
+    spec: {
+      podSelector: {
+        matchLabels: ksm._config.selectorLabels,
+      },
+      ingress: [{
+        from: [{
+          podSelector: {
+            matchLabels: {
+              'app.kubernetes.io/name': 'prometheus',
```

Comment:
Just like some recent changes you've made in #1471, maybe we could re-use labels from Prometheus object instead of hard-coding it here?

If someone that uses kube-prometheus has changed the common labels, then the network policies will break their stack 🤔.

Reply:
This label is hardcoded in prometheus-operator and users are not allowed to change it. Even operator is relying on the existence of that label - https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/statefulset.go#L306-L308